### PR TITLE
Log: Allow logfile perms to be set in sysconfig

### PR DESCRIPTION
**Pull request by chrissie-c**
See rhbz#1647136

### CODE REVIEW

**Comment by chrissie-c** [comment]
it seems we don't need any libqb changes :)

File: lib/common/logging.c [hunk 1]
```diff
@@ -239,16 +241,31 @@ crm_add_logfile(const char *filename)
...
+        }
+
         if (fix) {
             rc = fchown(logfd, pcmk_uid, pcmk_gid);
             if (rc < 0) {
                 crm_warn("Cannot change the ownership of %s to user %s and gid %d",
                          filename, CRM_DAEMON_USER, pcmk_gid);
             }
+	    if (!filemode) {
+                    filemode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP;
```
Comment (clumens) [inline lib/common/logging.c:261]: I think you could just initialize filemode to this when you first create the variable and then skip needing an extra block.

Comment (kgaillot) [reply lib/common/logging.c:261]: this avoids having to duplicate the default mask in the strtol() failure case though

Comment (kgaillot) [reply lib/common/logging.c:261]: though defining DEFAULT_LOGFILE_MASK could get around that

**Comment by chrissie-c** [comment]
New version force-push, thanks for the comments. I think I've got my indentation sorted out now

**Comment by kgaillot** [comment]
Thanks! :)

**Pull request merged**